packages/material-ui/src/Autocomplete/Autocomplete.js

```diff
@@ -205,20 +204,27 @@ export const styles = (theme) => ({
     [theme.breakpoints.up('sm')]: {
       minHeight: 'auto',
     },
-    '&[aria-selected="true"]': {
-      backgroundColor: theme.palette.action.selected,
-    },
-    '&[data-focus="true"]': {
+
+    '&$focused': {
       backgroundColor: theme.palette.action.hover,
     },
-    '&:active': {
+    '&$selected': {
       backgroundColor: theme.palette.action.selected,
+      '&$focused': {
+        backgroundColor: alpha(
+          theme.palette.action.selected,
+          theme.palette.action.selectedOpacity + theme.palette.action.hoverOpacity,
+        ),
+      },
     },
-    '&[aria-disabled="true"]': {
+    '&$disabled': {
       opacity: theme.palette.action.disabledOpacity,
       pointerEvents: 'none',
     },
```

Comment:
I'm not sure how styling should be updated. This follows ListItem component's styling.

https://github.com/mui-org/material-ui/blob/d9d93088bffc86b2dd06a2436c170e765b92462e/packages/material-ui/src/ListItem/ListItem.js#L26-L40